stdlib/Dates/src/types.jl

```diff
@@ -278,9 +281,12 @@ function Date(y::Int64, m::Int64=1, d::Int64=1)
     return Date(UTD(totaldays(y, m, d)))
 end
 
+const DATE_YEAR_TYPEMAX = 252522163911149
+const DATE_YEAR_TYPEMIN = -252522163911148
```

Comment:
FYI, I don't understand where the value came from, but the current implementation is `-25252522163911150`.

Reply:
well huh - I looked at the datetime one and saw the direction of asymmetry and pattern matched that to date, but that is indeed not what was there 🤦

Reply:
Yes, the values for `DateTime` are straightforward.
```julia-repl
julia> reinterpret(DateTime, typemax(Int64) >> 1)
146138513-04-24T15:36:27.903

julia> reinterpret(DateTime, typemin(Int64) >> 1)
-146138512-09-07T08:23:32.096
```

Reply:
xref: https://github.com/JuliaLang/julia/issues/50931